.github/workflows/node-hub-ci-cd.yml

```diff
@@ -18,12 +18,34 @@ jobs:
     outputs:
       folders: ${{ steps.jobs.outputs.folders }}
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v4
 
       - id: jobs
         uses: kmanimaran/list-folder-action@v4
         with:
           path: ./node-hub
+    
+  lint:
+    runs-on: ubuntu-24.04
+    needs: [find-jobs]
+    name: Lint
+    strategy:
+      fail-fast: ${{ github.event_name != 'workflow_dispatch' && !(github.event_name == 'release' && startsWith(github.ref, 'refs/tags/')) }}
+      matrix:
+        folder: ${{ fromJson(needs.find-jobs.outputs.folders )}}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install the latest version of uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+      
+      - name: Run linting
+        working-directory: node-hub/${{ matrix.folder }}
+        run: |
+          uv tool install ruff
+          ruff check .
```

Comment:
FYI, this is not necessary as we already have it in our node hub script:

https://github.com/dora-rs/dora/blob/1f35724f1541b13a4eba8be6cb78d09a2b73eb4d/.github/workflows/node_hub_test.sh#L89

:)